Malmo/src/MissionSpec.cpp

```diff
@@ -18,656 +18,625 @@
-        
-        xml_schema::namespace_infomap map;
-        map[""].name = xml_namespace;
-        map[""].schema = "Mission.xsd";
+        std::ostringstream oss;
 
-        xml_schema::flags flags = xml_schema::flags::dont_initialize;
-        if( !prettyPrint )
-            flags = flags | xml_schema::flags::dont_pretty_print;
+        write_xml(oss, mission);
 
-        Mission_( oss, *this->mission, map, "UTF-8", flags );
-        
-        return oss.str();
+        std::string xml = oss.str();
+        if (!prettyPrint) 
+            xml.erase(std::remove(xml.begin(), xml.end(), '\n'), xml.end());
+        return xml;
     }
     
     // -------------------- settings for the server ---------------------------------
     
     void MissionSpec::setSummary( const std::string& summary )
     {
-        this->mission->About().Summary( summary );
+        mission.put("Mission.About.Summary", summary);
     }
 
     void MissionSpec::timeLimitInSeconds(float s)
     {
-        ServerHandlers::ServerQuitFromTimeUp_sequence& qq = this->mission->ServerSection().ServerHandlers().ServerQuitFromTimeUp();
-        if( !qq.size() ) 
-        {
-            // make a new ServerQuitFromTimeUp handler for the server_handlers
-            ServerQuitFromTimeUp q( s * MillisecondsInOneSecond );
-            qq.push_back( q );
-        }
-        else
-        {
-            ServerQuitFromTimeUp& q = qq.front();
-            q.timeLimitMs( s * MillisecondsInOneSecond );
-        }
+        mission.put("Mission.ServerSection.ServerHandlers.ServerQuitFromTimeUp.<xmlattr>.timeLimitMs", s);
     }
     
     void MissionSpec::createDefaultTerrain()
     {
-        this->mission->ServerSection().ServerHandlers().FlatWorldGenerator().reset();
-        DefaultWorldGenerator dwg;
-        this->mission->ServerSection().ServerHandlers().DefaultWorldGenerator( dwg );
+        forceWorldReset();
+        mission.put("Mission.ServerSection.ServerHandlers.DefaultWorldGenerator","");
     }
 
     void MissionSpec::setWorldSeed(const std::string& seed)
     {
-        auto& default_wg = this->mission->ServerSection().ServerHandlers().DefaultWorldGenerator();
-        auto& flat_wg = this->mission->ServerSection().ServerHandlers().FlatWorldGenerator();
-        if (default_wg.present())
-            default_wg.get().seed(seed);
-        if (flat_wg.present())
-            flat_wg.get().seed(seed);
+        const auto& default_wg = mission.get_child_optional("Mission.ServerSection.ServerHandlers.DefaultWorldGenerator");
+        if (default_wg)
+            default_wg.get().put("<xmlattr>.seed", seed);
+        const auto& flat_wg = mission.get_child_optional("Mission.ServerSection.ServerHandlers.FlatWorldGenerator");
+        if (flat_wg)
+            flat_wg.get().put("<xmlattr>.seed", seed);
     }
 
     void MissionSpec::forceWorldReset()
     {
-        auto& default_wg = this->mission->ServerSection().ServerHandlers().DefaultWorldGenerator();
```

Comment:
forceWorldReset() exists to set the "forceReset" flag on the world generator - i.e so that each mission will start with a fresh world. Not to be confused with a method for _removing_ world generators!

Reply:
I misunderstood.Hopefully corrected now.